modules/apps/portal-search/portal-search-rest-impl/src/main/java/com/liferay/portal/search/rest/internal/resource/v1_0/SearchResultResourceImpl.java

```diff
@@ -0,0 +1,617 @@
+import com.liferay.portal.odata.entity.EntityModel;
+import com.liferay.portal.search.aggregation.AggregationResult;
+import com.liferay.portal.search.document.Document;
+import com.liferay.portal.search.document.Field;
+import com.liferay.portal.search.hits.SearchHit;
+import com.liferay.portal.search.hits.SearchHits;
+import com.liferay.portal.search.rest.configuration.HeadlessSearchAPICompanyConfiguration;
+import com.liferay.portal.search.rest.dto.v1_0.FacetConfiguration;
+import com.liferay.portal.search.rest.dto.v1_0.SearchRequestBody;
+import com.liferay.portal.search.rest.dto.v1_0.SearchResult;
+import com.liferay.portal.search.rest.internal.facet.FacetRequestContributor;
+import com.liferay.portal.search.rest.internal.facet.FacetResponseProcessor;
+import com.liferay.portal.search.rest.internal.odata.entity.v1_0.SearchResultEntityModel;
+import com.liferay.portal.search.rest.pagination.SearchPage;
+import com.liferay.portal.search.rest.resource.v1_0.SearchResultResource;
+import com.liferay.portal.search.searcher.SearchRequestBuilder;
+import com.liferay.portal.search.searcher.SearchRequestBuilderFactory;
+import com.liferay.portal.search.searcher.SearchResponse;
+import com.liferay.portal.search.searcher.Searcher;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.io.Serializable;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.apache.commons.lang.ArrayUtils;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Petteri Karttunen
+ */
+@Component(
+	configurationPid = "com.liferay.portal.search.rest.configuration.HeadlessSearchAPICompanyConfiguration",
+	properties = "OSGI-INF/liferay/rest/v1_0/search-result.properties",
+	scope = ServiceScope.PROTOTYPE, service = SearchResultResource.class
+)
+public class SearchResultResourceImpl extends BaseSearchResultResourceImpl {
+
+	@Override
+	public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
+		return _searchResultEntityModel;
+	}
+
+	@Override
+	public Page<SearchResult> postSearchPage(
+			String documentFields, String entryClassNames, String groupIds,
+			String search, Filter filter, Pagination pagination, Sort[] sorts,
+			SearchRequestBody searchRequestBody)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPS-179669")) {
+			return null;
```

Comment:
Here it is better to throw a NotFoundException so that it behaves as it the endpoint did not exist at all.